components/dashboard/src/onboarding/StepOrgInfo.tsx

```diff
@@ -0,0 +1,180 @@
+                    companyWebsite,
+                },
+            },
+        };
+
+        try {
+            const updatedUser = await updateUser.mutateAsync(updates);
+            onComplete(updatedUser);
+        } catch (e) {
+            console.error(e);
+        }
+    }, [
+        companyWebsite,
+        jobRole,
+        jobRoleOther,
+        onComplete,
+        signupGoals,
+        signupGoalsOther,
+        updateUser,
+        user.additionalData,
+    ]);
+
+    const jobRoleError = useOnBlurError("Please select one", !!jobRole);
+    const isValid = jobRoleError.isValid && signupGoals.length > 0;
+
+    return (
+        <OnboardingStep
+            title="Tell us more about you"
+            subtitle="Let us know what brought you here."
+            error={updateUser.isError ? "There was a problem saving your answers" : ""}
+            isValid={isValid}
+            isLoading={updateUser.isLoading}
+            onSubmit={handleSubmit}
+        >
+            <SelectInputField
+                value={jobRole}
+                label="I work in..."
+                onChange={(val) => {
+                    if (val !== "other") {
+                        setJobRoleOther("");
+                    }
+                    setJobRole(val);
+                }}
+                hint={
+                    jobRole !== JOB_ROLE_OTHER
+                        ? "Please select the role that best describes the type of work you'll use Gitpod for"
+                        : ""
+                }
+                error={jobRoleError.message}
+                onBlur={jobRoleError.onBlur}
+            >
+                {jobRoleOptions.map((o) => (
+                    <option key={o.value} value={o.value}>
+                        {o.label}
+                    </option>
+                ))}
+            </SelectInputField>
+
+            {jobRole === JOB_ROLE_OTHER && (
+                <TextInputField
+                    value={jobRoleOther}
+                    onChange={setJobRoleOther}
+                    placeholder="Please specify"
+                    hint={
+                        jobRole === JOB_ROLE_OTHER
+                            ? "Please select the role that best describes the type of work you'll use Gitpod for"
+                            : ""
+                    }
+                />
+            )}
+
+            <TextInputField
+                value={companyWebsite}
+                label="Company Website (optional)"
+                type="url"
+                placeholder="https://"
+                onChange={setCompanyWebsite}
+            />
+
+            <InputField label="I'm signing up for Gitpod to..." />
+            <div className="mt-4 ml-2 space-y-2">
+                {signupGoalsOptions.map((o) => (
+                    <div key={o.value} className="flex space-x-2 justify-start items-center">
+                        <input
+                            type="checkbox"
```

Comment:
**suggestion**: We do have a [`Checkbox`](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/src/components/CheckBox.tsx) component that could be reused here.

Reply:
I need to make one, this was the first checkbox I've added ui for, so wanted to build it out a little, and will extract it to a component in a followup. Checkboxes are always so different than other inputs, so I wasn't sure what our UX pattern might look like for them.

Reply:
DEAL—Opened https://github.com/gitpod-io/gitpod/issues/16511 yesterday that mentions a checkbox component which is possibly worth splitting into a separate issues for each form element. 🏓